### DISCUSSION TITLE
CVE-2023-41106

### DISCUSSION THREAD

**Discussion by ianw1974** [ianw1974/zimbra-build-scripts#43] [General]
Hi,

New builds are currently being made and will appear on my site shortly.  An issue was opened here relating to this: #41 

The vulnerability is within Zimbra itself, not within the scripts.  However, I felt it important to address this via discussion as well as an issue so that people are aware that I am working on this.

Please feel free to post to discuss this.  I will attempt to compare two systems before and after to find out what changes if any have addressed it.

**Comment by ianw1974** [comment]
New builds are now up.  I'll be looking to figure out shortly, whether the problem has been addressed

**Comment by ianw1974** [comment]
After comparing the previous builds with the new ones from yesterday evening, the issue wasn't addressed.  This has now been rectified and new builds are in progress.

**Comment by ianw1974** [comment]
New builds now available, problem is resolved.  Checked/tested a build to ensure on a VPS for clean install as well as upgrade from previous build.

**Reply by alepaes1975** [reply]
Hi Ian,

I installed the last version packaged for you (10.0.0_GA_0001.UBUNTU20_64.20230824161408.tgz), but the version reported in zmcontrol -v is 10.0.0 and not 10.0.3. I enable zimbra repo in installation, but the patch is noy avaiable. Thats is the correct way ?

**Comment by ianw1974** [comment]
You don't need the patch, because I remove the files from the source at build time.  If you check for the files that are supposed to be removed, you will find that they are gone.

**Reply by alepaes1975** [reply]
Ahh ok, thanks!

**Reply by ianw1974** [reply]
As for the versioning, mine are in line with for example like here: https://www.zimbra.com/product/download/zimbra-collaboration-network-edition/

If you check the filenames, you will see they show 10.0.0.  Which is the same as mine.  I could maybe change in the config.build file to 10.0.3 and see if that also reflects in the webui, but I'm not entirely sure of the consequences of doing that.  For example, will the build process fail, or continue and the value in config.build being purely cosmetic.  I don't generally know when Zimbra will release a particular version/point release, and could in theory just increase this myself when I build each quarter, but then it wouldn't be in sync with Zimbra.

I could always increase the build number, to example day/month the build was made, but that would purely be versioning that I personally would introduce to differentiate from my earlier builds.

Something I could think about doing, but generally so far I've always been in sync with the way Zimbra releases, so still using 10.0.0 which is reflected in the filenames.

**Reply by alepaes1975** [reply]
In the current Zimbra method, the update reflects the patch version when doing a -v (i.e.: Release 8.8.15.GA.3869.UBUNTU18.64 UBUNTU18_64 FOSS edition, Patch 8.8.15_P40). I have scripts the check to inform us if a version is outdated.

**Reply by ianw1974** [reply]
Yep, the biggest problem here is:

1. Knowing when they are going to put 8.8.15 instead of 8.8.0 for example.
2. Knowing what build number they are going to use, 3869 in this example
3. The P40 will most likely come from installing their patch package.

Things pretty much are impossible for me to know about until Zimbra actually release and difficult if not impossible to integrate into builds.  I do not provide repositories, Zimbra's install will still use them, but whether that is also available for free users than paying users is also another issue (there was talk of these patches etc not being available to free users).  So many things totally out of my control.

I can provide builds pretty much and that is all.  If the numbers do not match, that's something thats going to be hard to keep in sync with.  More important is, that I address the issues and ensure that they have been applied in my builds.

Zimbra's repository still builds and provides the files that were mentioned in the CVE.  Therefore, I remove it before building because otherwise, my new builds would still have them included - just like Zimbra's builds for Network Edition download also still include them because they are old builds from March.

That's pretty much all I can do to ensure that something has been fixed by ensuring my build scripts deal with this, as well as me utilising them to provide my builds for download.

**Reply by alepaes1975** [reply]
I understood. Thanks for your work !
